app/controllers/annotations_controller.rb

```diff
@@ -144,7 +152,11 @@ def update_annotation
     @content = params[:annotation_text][:content]
     @id = params[:annotation_text][:id]
     @submission_file_id = params[:submission_file_id]
```

Comment:
Please remove this line. `@submission_file` should be populated by `@annotation.submission_file`. You can make this change in `destroy` as well.

By doing so, you can eliminate the need to pass in the `submission_file_id` in the AJAX request.